### ISSUE TITLE
Change Request: Add "test name" option to ValidTestCase / InvalidTestCase

### ISSUE THREAD

**Issue by ilyub** [eslint/eslint#15090]
### ESLint version

7.32.0

### What problem do you want to solve?

When there are many tests it is easier to find failed test by name than by code.

### What do you think is the correct solution?

Add "test name" option to ValidTestCase / InvalidTestCase

### Participation

- [ ] I am willing to submit a pull request for this change.

### Additional comments

Related issue:
https://github.com/typescript-eslint/typescript-eslint/issues/3920

**Comment by nzakas** [comment]
> Add "test name" option to ValidTestCase / InvalidTestCase

How would this work? Are you expecting that the name would be output instead of the code or in addition to it?

I’m not opposed to this but it seems unlikely people would use it very often as rules can have a lot of tests and it’s often hard to come up with descriptive names for each test.

**Comment by ilyub** [comment]
Thx for reply.

> How would this work? Are you expecting that the name would be output instead of the code or in addition to it?

I would prefer the name (if present) to be output instead of the code. This would make  output more compact.

> it seems unlikely people would use it very often

1:
The name can be optional. So user can omit it if he does not find it useful.

2:
If you look at Jest syntax you will see that they have name and it is even required:
```ts
it("The name of the test", () => {
  // Test code
});
```

3:
From my expirience, when you have many tests it is quite difficult to find failed test by code. So, I end up adding test name as comment:
```ts
[
    {
      code: `
        // Test 1
        ...
     `
   },
   {
      code: `
        // Test 2
        ...
     `
   },
```

> it’s often hard to come up with descriptive names for each test.

This is true, but I am just adding number "Test 1", "Test 2", etc.
But I have to add it as a comment.

---

By the way, I would also suggest "only" option.

I find this option both in Jest and in typescript/eslint project and it is very convenient. It helps to quickly disable all tests but one without commenting out other tests.

I already have it because I am using typescript/eslint, but those people who write directly for eslint will apreciate it.

**Comment by mdjermanovic** [comment]
> By the way, I would also suggest "only" option.
> 
> I find this option both in Jest and in typescript/eslint project and it is very convenient. It helps to quickly disable all tests but one without commenting out other tests.
> 
> I already have this option because I am using typescript/eslint, but those people who write directly for eslint will apreciate it.

This option already exist: https://eslint.org/docs/developer-guide/unit-tests#running-individual-tests

**Comment by ilyub** [comment]
Sorry, probably "only" option was added after I switched to typescript/eslint.

But "name" option is still wanted.

**Comment by nzakas** [comment]
I’m not opposed to adding a name option. Would you be interested in implementing it if we can point you to the code?

**Comment by ilyub** [comment]
I tried to follow:
https://eslint.org/docs/developer-guide/development-environment

But when I run "npm test" I get this:
![Screenshot](https://user-images.githubusercontent.com/70687432/134821083-da322a47-9fff-4068-8a06-555f120d46d0.jpg)
and "coverage" folder is not created. So, testing is not done.

If this can be easily solved, I could try to make an edit.

Though I would prefer this to be done by somebody familiar with this project.

**Comment by nzakas** [comment]
@ilyub I'm unable to reproduce this with the master branch. I'd suggest re-syncing with the master branch, removing `node_modules` folder, and re-installing.

https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/syncing-a-fork

**Comment by G-Rath** [comment]
I've made #15179 implementing this
